libraries/AP_HAL_Linux/TCPClientDevice.cpp

```diff
@@ -0,0 +1,65 @@
+#include <AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <stdlib.h>
+
+#include "TCPClientDevice.h"
+
+TCPClientDevice::TCPClientDevice(const char *ip, uint16_t port):
+    _ip(ip),
+    _port(port)
+{
+}
+
+TCPClientDevice::~TCPClientDevice()
+{
+
+}
+
+int TCPClientDevice::write(const uint8_t *buf, uint16_t n)
+{
+    return listener.send(buf, n);
```

Comment:
if we were going to support a TCP client device then I don't think calling it "listener" is a good name. listen() is for servers in TCP